virttest/qemu_vm.py

```diff
@@ -1869,27 +1887,77 @@ def sort_key(dev):
                 txt = "Set vcpu_threads to 1 for AMD cpu."
                 logging.warn(txt)
 
-        if smp == 0 or vcpu_sockets == 0:
-            vcpu_cores = vcpu_cores or 1
-            vcpu_threads = vcpu_threads or 1
-            if smp and vcpu_sockets == 0:
-                vcpu_sockets = int(smp / (vcpu_cores * vcpu_threads)) or 1
-            else:
-                vcpu_sockets = vcpu_sockets or 1
-            if smp == 0:
-                smp = vcpu_cores * vcpu_threads * vcpu_sockets
+        vcpu_devices = params.objects('vcpu_devices')
```

Comment:
Should we distinguish cpu devices if they are added to qemu cli, or they will be hotplugged later?

Reply:
> Should we distinguish cpu devices if they are added to qemu cli, or they will be hotplugged later?

Whether it is added to the command lines or prepared for hot plugging later, they belong to the cpu device. It will be supported in the next patch and differentiated using `vcpu_enabled`